lib/tasks/bundlers/generateLibraryManifest.js

```diff
@@ -0,0 +1,60 @@
+"use strict";
+
+const log = require("@ui5/logger").getLogger("builder:tasks:bundlers:generateLibraryManifest");
+const manifestCreator = require("../../processors/manifestCreator");
+const ReaderCollectionPrioritized = require("@ui5/fs").ReaderCollectionPrioritized;
+
+
+/**
+ * Task for creating a library manifest.json from its .library file.
+ *
+ * @module builder/tasks/bundlers/generateLibraryManifest
+ * @param {Object} parameters Parameters
+ * @param {DuplexCollection} parameters.workspace DuplexCollection to read and write files
+ * @param {AbstractReader} parameters.dependencies Reader or Collection to read dependency files
+ * @param {Object} parameters.options Options
+ * @param {string} parameters.options.projectName Project name
+ * @returns {Promise<undefined>} Promise resolving with <code>undefined</code> once data has been written
+ */
+module.exports = function({workspace, dependencies, options}) {
+	const combo = new ReaderCollectionPrioritized({
+		name: `libraryManifestGenerator - prioritize workspace over dependencies: ${options.projectName}`,
+		readers: [workspace, dependencies]
+	});
+	return combo.byGlob("/**/*.{js,json,xml,html,properties,library,less,css,theming,theme}").then((resources) => {
+		// Find all libraries and create a manifest.json file
+		return workspace.byGlob("/resources/**/.library").then((libraryIndicatorResources) => {
+			if (libraryIndicatorResources.length < 1) {
+				// No library found - nothing to do
+				log.error(`Could not find a ".library" file for project ${options.projectName}. Skipping library manifest generation.`);
```

Comment:
Do we want this to be an error?
I think most of the libraries developed in the wild doesn't have a `.library` file as it wasn't used at all except by the maven tooling.

Reply:
If we take "manifest first for libraries" serious, we need at least to enforce a `manifest.json`. So we could check for that and if there's a `manifest.json` then the `.library` becomes optional.

But it doesn't have to be this task that complains about it.

Reply:
More likely the job of the [LibraryFormatter](https://github.com/SAP/ui5-builder/blob/master/lib/types/library/LibraryFormatter.js#L61).

The ApplicationFormatter already [reads an applications manifest](https://github.com/SAP/ui5-builder/blob/master/lib/types/application/ApplicationFormatter.js#L16), though it doesn't insist on it.